libminifi/include/utils/MinifiConcurrentQueue.h

```diff
@@ -99,6 +124,20 @@ class ConcurrentQueue {
     return true;
   }
 
+  template<typename Functor>
+  bool consumeImpl(std::unique_lock<std::mutex>&& lck, Functor&& fun) {
+    static_assert(std::is_nothrow_move_assignable<T>::value, "T has to be nothrow move assignable.");
```

Comment:
This is the one only doing move construction (once for initializing `elem` and potentially a second time when moving elem to the `fun` argument) and the other one is only doing move assignment (when assigning the element to `out`).

Reply:
Corrected.